src/order/liminf_limsup.lean

```diff
@@ -540,8 +607,109 @@ lemma le_limsup_of_frequently_le' {α β} [complete_lattice β]
   x ≤ f.limsup u :=
 @liminf_le_of_frequently_le' _ βᵒᵈ _ _ _ _ h
 
+variables {f : filter β} {p q : β → Prop} {u v : β → α}
+
+lemma blimsup_mono (h : ∀ x, p x → q x) :
+  f.blimsup p u ≤ f.blimsup q u :=
+Inf_le_Inf $ λ a ha, ha.mono $ by tauto
+
+lemma bliminf_antitone (h : ∀ x, p x → q x) :
+  f.bliminf q u ≤ f.bliminf p u :=
+Sup_le_Sup $ λ a ha, ha.mono $ by tauto
+
+lemma mono_blimsup' (h : ∀ᶠ x in f, u x ≤ v x) :
+  f.blimsup p u ≤ f.blimsup p v :=
+Inf_le_Inf $ λ a ha, (ha.and h).mono $ λ x hx hx', hx.2.trans (hx.1 hx')
+
+lemma mono_blimsup (h : ∀ x, u x ≤ v x) :
+  f.blimsup p u ≤ f.blimsup p v :=
+mono_blimsup' $ eventually_of_forall h
+
+lemma mono_bliminf' (h : ∀ᶠ x in f, u x ≤ v x) :
+  f.bliminf p u ≤ f.bliminf p v :=
+Sup_le_Sup $ λ a ha, (ha.and h).mono $ λ x hx hx', (hx.1 hx').trans hx.2
+
+lemma mono_bliminf (h : ∀ x, u x ≤ v x) :
+  f.bliminf p u ≤ f.bliminf p v :=
+mono_bliminf' $ eventually_of_forall h
```

Comment:
Is there also a monotonicity with respect to the filter?

Reply:
Good point. I have just added two lemmas for this. The naming was difficult because `blimsup` is a three-variable function and I've already used up `mono_blimsup` and `blimsup_mono` for the other two variables so I went with `blimsup_monotone_filter` (likewise for `blimsup`).